input.go

```diff
@@ -108,10 +108,10 @@ func InputsSyntacticalIndicesWithinBounds() ElementValidationFunc[Input] {
 		case *UTXOInput:
 			// TODO: do we really want to check the max value on the input side?
 			if castInput.Index() < RefUTXOIndexMin || castInput.Index() > RefUTXOIndexMax {
-				return ierrors.Wrapf(ErrRefUTXOIndexInvalid, "input %d", index)
+				return ierrors.WithMessagef(ErrRefUTXOIndexInvalid, "input %d", index)
 			}
 		default:
-			return ierrors.Wrapf(ErrUnknownInputType, "input %d, tx can only contain UTXInput inputs", index)
+			panic("all supported input types should be handled above")
```

Comment:
here as well.

Reply:
Btw just above this line I think you added a TODO: 
> // TODO: do we really want to check the max value on the input side?

I'm not fully sure what you mean by it, but if you mean why we check that the index is within bounds on the input side even though we only create indicies within that range on the output side: the index on the `UTXOInput` can be set by a user so it could be set to whatever value, so we should check here. Although not checking wouldn't be problematic either, because we just would run into a "input not found" error later.

Reply:
We discussed that in the past, and we want to actually have that property, that we can inject bigger indexes in the snapshot, but the user should not be able to create it in a transaction. So we check it on the output side, but not on the input side.
As you said, we simply would run into "input not found" if a user crafts a transaction with a higher index that doesn't exist.